### ISSUE TITLE
error building windows template with mingw

### ISSUE THREAD

**Issue by romulox-x** [godotengine/godot#3024]
drivers/unix/os_unix_global_settings_path.cpp:2:1: error: 'String' does not name a type

when building with:
scons platform=windows tools=no target=release_debug bits=64 -j 8

**Comment by vnen** [comment]
I have the same problem building for Windows on Arch Linux.

**Comment by purpleorangegames** [comment]
Same error building for windows on linux (ubuntu 64 on digital ocean) on 08/Dec/2015:
[20:00:26]Step 1/28: Compile Linux 64 Server (Command Line) (31m:24s)
[20:31:50]Step 2/28: Clear Linux 64 Server (Command Line) (1m:44s)
[20:33:35]Step 3/28: Compile Windows 32 Tool Theora (Command Line) (1m:59s) failed
[20:35:35]Step 4/28: Clear Windows 32 Theora (Command Line)
[20:35:35]Step 5/28: Compile Linux 64 Debug (Command Line) (13m:18s)
[20:48:53]Step 6/28: Compile Linux 64 Tool (Command Line) (33m)
[21:21:54]Step 7/28: Compile Linux 64 Release (Command Line) (23m:18s)
[21:45:12]Step 8/28: Clear Linux 64 (Command Line) (10m:14s)
[21:55:26]Step 9/28: Compile Linux 32 Debug (Command Line) (13m:06s)
[22:08:33]Step 10/28: Compile Linux 32 Tool (Command Line) (33m:37s)
[22:42:11]Step 11/28: Compile Linux 32 Release (Command Line) (23m:20s)
[23:05:31]Step 12/28: Clear Linux 32 (Command Line) (8m:59s)
[23:14:30]Step 13/28: Compile Windows 32 Debug (Command Line) (1m:12s) failed
[23:15:42]Step 14/28: Compile Windows 32 Tool (Command Line) (6s) failed
[23:15:49]Step 15/28: Compile Windows 32 Release (Command Line) (1m:47s) failed
[23:17:37]Step 16/28: Clear Windows 32 (Command Line)
[23:17:37]Step 17/28: Compile Windows 64 Debug (Command Line) (1m:12s) failed
[23:18:50]Step 18/28: Compile Windows 64 Tool (Command Line) (1m:54s) failed
[23:20:44]Step 19/28: Compile Windows 64 Release (Command Line) (1m:43s) failed
[23:22:28]Step 20/28: Clear Windows 64 (Command Line)
[23:22:28]Step 21/28: Compile OSX 64 Debug (Command Line) (18m:14s)
[23:40:42]Step 22/28: Compile OSX 64 Release (Command Line) (23m:06s)
[00:03:48]Step 23/28: Compile OSX 64 Tool (Command Line) (41m:27s)
[00:45:16]Step 24/28: Clear OSX 64 (Command Line) (12m:47s)
[00:58:03]Step 25/28: Compile OSX 32 Debug (Command Line) (18m:40s)
[01:16:44]Step 26/28: Compile OSX 32 Release (Command Line) (23m:35s)
[01:40:19]Step 27/28: Compile OSX 32 Tool (Command Line) (41m:45s)
[02:22:05]Step 28/28: Clear OSX 32 (Command Line) (11m:43s)

**Comment by purpleorangegames** [comment]
Looking at the commits I can see that they changed how to detect Linux systems, the file drivers/unix/os_unix_global_settings_path.cpp is automatically generated for linux systems:

```
changed a bit to use system()	reduzio		1 file 	708ff381d6c7	08 Dec 15 18:36	  
improve detection of linux, fixes #3201	reduzio		1 file 	78b5404434ee	08 Dec 15 18:35	
```
From
```
if platform.platform() == 'Linux':
```
To
```
if platform.platform().find('Linux')>=0:
```
Then To
```
if platform.system() == 'Linux':
```

IN drivers/unix/SCsub you can see that it is unix only:
```
ed_gl_set+='}\n'
f = open("os_unix_global_settings_path.cpp","wb")
f.write(ed_gl_set)
f.close()
	
env.add_source_files(env.drivers_sources,"*.cpp")
	
Export('env')
```
You can also find it's name in .gitignore

So even if I'm compiling it for windows, it is trying to compile linux drivers or something.

On a sucessful build I have this log:
```
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/mutex_posix.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/thread_posix.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/memory_pool_static_malloc.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/os_unix.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/stream_peer_tcp_posix.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/ip_unix.windows.opt.tools.32.o
```
And on a new build:
```
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/mutex_posix.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/thread_posix.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/memory_pool_static_malloc.windows.opt.tools.32.o
[Step 3/28] i686-w64-mingw32-g++ -o drivers/unix/os_unix_global_settings_path.windows.opt.tools.32.o
[Step 3/28] drivers/unix/os_unix_global_settings_path.cpp:2:1: error: 'String' does not name a type
[Step 3/28]  String OS_Unix::get_global_settings_path() const {
```

**Comment by akien-mga** [comment]
> Unless os_unix_global_settings_path was introduced recently and is the problem itself.

That is the case, it was added in #2956.

The changes to Linux detection in platform/x11/detect.py are not relevant for this bug, as you can see the ```if``` only handles the linking of the ALSA library on Linux.

**Comment by purpleorangegames** [comment]
Alright! Great to know what introduced this behaviour.

**Comment by akien-mga** [comment]
ping @est31

**Comment by vnen** [comment]
Fixed by #3026.
